Doctor cannot distinguish test timeout from test failure

## Bug

When the shepherd routes to the Doctor phase after a test timeout, the Doctor treats it the same as a test failure and tries to fix the code. Since the code is correct (it's a timeout, not a failure), the Doctor exits with `insufficient_changes` and the cycle repeats uselessly.

## Observed Behavior

```
[ERROR] Tests timed out after 300s (pytest)
[WARN] Builder test verification failed (883s), routing to Doctor (attempt 1/2)
...
[WARN] Doctor test-fix failed (doctor test-fix failed with exit code 1 (insufficient_changes)), re-running test verification
...
[ERROR] Tests timed out after 300s (pytest)
[WARN] Builder test verification failed (337s), routing to Doctor (attempt 2/2)
...
[ERROR] Builder test verification failed after 2 Doctor fix attempt(s) (1556s)
```

The Doctor spent ~10 minutes per attempt examining correct code, then exited with `insufficient_changes` because there was nothing to fix. The verification then timed out again.

## Expected Behavior

The shepherd should distinguish between:
1. **Test timeout** — Don't route to Doctor. Instead, either increase timeout, scope tests, or report the timeout directly to the user.
2. **Test failure** — Route to Doctor to fix the failing tests.

## Suggested Fix

In the shepherd's test verification result handling:
- If exit reason is `timeout`, skip doctor and report: "Tests timed out after Ns. Consider increasing test timeout or using scoped tests."
- If exit reason is `failure` (non-zero exit with actual test output), route to Doctor as usual.

## Impact

Wasted ~25 minutes on two futile Doctor cycles that could never succeed because the problem was a timeout configuration issue, not a code defect.

## Environment

- Loom Version: 0.2.3
- Repository: rjwalters/kicad-tools

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -515,6 +515,35 @@ def orchestrate(ctx: ShepherdContext) -> int:
             if result.is_shutdown:
                 raise ShutdownSignal(result.message)
 
+            # Handle test timeouts — skip Doctor since the code is likely correct
+            if (
+                result.status in (PhaseStatus.FAILED, PhaseStatus.STUCK)
+                and result.data.get("test_timeout")
+            ):
+                log_error(
+                    f"Test verification timed out. Skipping Doctor — "
+                    f"timeouts indicate a resource/configuration issue, not a code defect. "
+                    f"Consider increasing test timeout or using scoped tests."
+                )
+                phase_durations["Builder"] = builder_total_elapsed
+                ctx.report_milestone(
+                    "phase_completed",
+                    phase="builder",
+                    duration_seconds=builder_total_elapsed,
+                    status="test_timeout",
+                )
+                _mark_builder_test_failure(ctx)
+                _run_reflection(
+                    ctx,
+                    exit_code=ShepherdExitCode.PR_TESTS_FAILED,
+                    duration=int(time.time() - start_time),
+                    phase_durations=phase_durations,
+                    completed_phases=completed_phases,
+                    test_fix_attempts=0,
+                    warnings=run_warnings,
+                )
+                return ShepherdExitCode.PR_TESTS_FAILED
+
             # Handle test failures with Doctor test-fix loop
             while (
                 result.status in (PhaseStatus.FAILED, PhaseStatus.STUCK)

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -1587,7 +1587,7 @@ class BuilderPhase:
                     status=PhaseStatus.FAILED,
                     message=f"supplemental test timed out after {elapsed}s ({display_name})",
                     phase_name="builder",
-                    data={"test_failure": True},
+                    data={"test_failure": True, "test_timeout": True},
                 )
             except OSError as e:
                 log_warning(f"Could not run supplemental test ({display_name}): {e}")
@@ -1918,7 +1918,7 @@ class BuilderPhase:
                 status=PhaseStatus.FAILED,
                 message=f"test verification timed out after {elapsed}s ({display_name})",
                 phase_name="builder",
-                data={"test_failure": True},
+                data={"test_failure": True, "test_timeout": True},
             )
         except OSError as e:
             log_warning(f"Could not run tests ({display_name}): {e}")
@@ -2104,7 +2104,7 @@ class BuilderPhase:
                 status=PhaseStatus.FAILED,
                 message=f"test verification timed out after {elapsed}s ({display_name})",
                 phase_name="builder",
-                data={"test_failure": True},
+                data={"test_failure": True, "test_timeout": True},
             )
         except OSError as e:
             log_warning(f"Could not run tests ({display_name}): {e}")

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -2517,6 +2517,70 @@ class TestDoctorTestFixLoop:
         assert result == ShepherdExitCode.SHUTDOWN
 
 
+class TestTestTimeoutSkipsDoctor:
+    """Test that test timeouts skip the Doctor loop (issue #2391)."""
+
+    @patch("loom_tools.shepherd.cli._run_reflection")
+    @patch("loom_tools.shepherd.cli._mark_builder_test_failure")
+    @patch("loom_tools.shepherd.cli.DoctorPhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_timeout_skips_doctor(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockDoctor: MagicMock,
+        mock_mark_failure: MagicMock,
+        mock_reflection: MagicMock,
+    ) -> None:
+        """Test timeout returns PR_TESTS_FAILED without invoking Doctor."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            5,     # builder phase_start
+            305,   # builder elapsed (timeout)
+            305,   # duration calc in reflection
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+        ctx.worktree_path = Path("/fake/repo/.loom/worktrees/issue-42")
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (False, "")
+        # Builder reports test timeout (not just failure)
+        builder_inst.run.return_value = PhaseResult(
+            status=PhaseStatus.FAILED,
+            message="test verification timed out after 300s (pytest)",
+            phase_name="builder",
+            data={"test_failure": True, "test_timeout": True},
+        )
+
+        result = orchestrate(ctx)
+        assert result == ShepherdExitCode.PR_TESTS_FAILED
+
+        # Doctor should NOT have been called
+        doctor_inst = MockDoctor.return_value
+        doctor_inst.run_test_fix.assert_not_called()
+        # Should be marked as test failure
+        mock_mark_failure.assert_called_once()
+        # Reflection should report test_timeout status
+        ctx.report_milestone.assert_any_call(
+            "phase_completed",
+            phase="builder",
+            duration_seconds=300,
+            status="test_timeout",
+        )
+
+
 class TestDoctorRegressionGuard:
     """Test that the doctor test-fix loop aborts when doctor makes things worse."""
 

